catalogd/internal/source/containers_image.go

```diff
@@ -296,6 +293,10 @@ func (i *ContainersImageRegistry) unpackImage(ctx context.Context, unpackPath st
 		return wrapTerminal(fmt.Errorf("catalog image is missing the required label %q", ConfigDirLabel), specIsCanonical)
 	}
 
+	// ensure unpack directory is empty
+	if err := os.RemoveAll(unpackPath); err != nil {
+		return fmt.Errorf("error removing unpacked path: %w", err)
+	}
 	if err := os.MkdirAll(unpackPath, 0700); err != nil {
 		return fmt.Errorf("error creating unpack directory: %w", err)
 	}
```

Comment:
Another place for "ensureEmptyDirectory"? So maybe make that function a utility that we can reuse in a few places?